user/coreutils/mkfsfat32.c

```
@@ -0,0 +1,316 @@
+    bpb->ext_flags = 0;
+    bpb->fs_version = 0;
+    bpb->root_cluster = root_cluster;
+    bpb->fsinfo_sector = 1;
+    bpb->backup_boot_sector = 6;
+    bpb->drive_number = 0x80;
+    bpb->boot_sig = 0x29;
+    bpb->volume_id = (uint32)(get_ticks() ^ (uint64)getpid());
+    fill_label(bpb->volume_label, label);
+    memcpy(bpb->fs_type, "FAT32   ", 8);
+
+    if (write_sector(dev, sector_size, 0, sector) < 0) return -1;
```

Cppcheck (2.20.0)

[error] Memory leak

(memleak)